packages/components/src/FormField/FormField.tsx

```diff
@@ -146,145 +138,150 @@ export interface FormFieldProps {
   onBlur?(): void;
 
   /**
-   * **DEPRECATED** Use `onValidation` prop instead.
-   *
    * Callback to get the the status and message when validating a field
-   * @param status
-   * @param message
+   * @param messages
    */
-  onValidate?(status: "pass" | "fail", message: string): void;
+  onValidation?(message: string | undefined): void;
 
   /**
-   * **EXPERIMENTAL** This feature is still under development.
-   *
-   * Callback to get the the status and message when validating a field
-   * @param messages
+   * Show an error message above the field. This also
+   * highlights the the field red if an error message shows up.
    */
-  onValidation?(messages: ValidationProps[]): void;
+  readonly validations?: ValidationRules;
+
+  inputRef?: RefObject<
+    HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+  >;
 }
 
-export const FormField = React.forwardRef(
-  /**
-   * Max statements is disabled for this component due the event handlers.
-   */
-  // eslint-disable-next-line max-statements
-  (
+/**
+ * Max statements is disabled for this component due the event handlers.
+ */
+// eslint-disable-next-line max-statements
+export function FormField({
+  align,
+  autocomplete = true,
+  children,
+  defaultValue,
+  disabled,
+  inline,
+  invalid,
+  max,
+  maxLength,
+  min,
+  name,
+  onFocus,
+  onBlur,
+  onChange,
+  onEnter,
+  onValidation,
+  placeholder,
+  readonly,
+  rows,
+  size,
+  type = "text",
+  value,
+  validations,
+  inputRef,
+}: FormFieldProps) {
+  const {
+    register,
+    /**
+     * We currently only use `errors` from formState, but there are lots of
+     * other values that come with it. https://react-hook-form.com/api#formState
+     */
+    formState: { errors },
+  } =
+    useFormContext() != undefined
+      ? useFormContext()
+      : useForm({ mode: "onBlur" });
```

Comment:
What do you think about `onTouched` here instead? I _think / hope_ this would mean that it is `onBlur` the first time then `onChange` for subsequent changes?